dask/array/overlap.py

```diff
@@ -841,3 +846,97 @@ def coerce_boundary(ndim, boundary):
     if isinstance(boundary, dict):
         boundary = {ax: boundary.get(ax, default) for ax in range(ndim)}
     return boundary
+
+
+def _window_view(arr, window_shape, axis, fix_first_block, block_info):
+    """ core sliding window function"""
+    from .numpy_compat import sliding_window_view
+
+    block_info = block_info[0]
+    slicer = [slice(None, None)] * arr.ndim
+    for ax, window, fix_first in zip(axis, window_shape, fix_first_block):
+        start, stop = None, None
+        is_last_chunk = (
+            block_info["num-chunks"][ax] - 1 - block_info["chunk-location"][ax]
+        ) == 0
+
+        locs = block_info["array-location"][ax]
+        size = locs[1] - locs[0]
+        if not is_last_chunk:
+            stop = -1 * max(window // 2 - 1, 1)
+
+        # when depth == chunk size of first chunk along some axis
+        # we need to make some confusing empirical adjustments.
+        # could be avoided if we rechunked so that min
+        # chunk size is depth+1 (which would also be more efficient)
+        # 1) don't trim the first block along that axis
+        if size == window and block_info["chunk-location"][ax] == 0:
+            stop = None
+        # 2) we need to skip the first "row" of the second block
+        if fix_first and block_info["chunk-location"][ax] == 1:
+            start = 1
+        slicer[ax] = slice(start, stop)
+
+    view = sliding_window_view(arr[tuple(slicer)], window_shape, axis)
+    return view
+
+
+def trim_chunks(chunks, trims):
+    """Trim chunk sizes at the end.
+
+    Examples
+    --------
+    >>> trim_chunks(((2, 1), (3, 2), (4, 1)), (0, 3, 2))
+    ((2, 1), (2,), (3,))
+    >>> trim_chunks(((5, 5), (4, 4)), (5, 4))
+    ((5,), (4,))
+    """
+    assert len(chunks) == len(trims)
+    newchunks = list(chunks)
+    for ax, trim in zip(range(len(chunks)), trims):
+        if trim == 0:
+            continue
+        c0 = np.cumsum(chunks[ax][::-1])[::-1]
+        idx = c0 >= (trim)
+        lastidx = np.nonzero(idx)[0][-1]
+        newchunks[ax] = chunks[ax][:lastidx]
+        if c0[lastidx] > trim:
+            newchunks[ax] += (c0[lastidx] - trim,)
+    return tuple(newchunks)
+
+
+def sliding_window_view(arr, window_shape, axis):
```

Comment:
```suggestion
def sliding_window_view(arr, window_shape, axis=None):
""" Dask wrapper for numpy.lib.stride_tricks.sliding_window_view."""

axis = range(arr.ndim) if axis is None
```

the numpy function specifies a default argument for `axis`, which applies the window along all axes. Not sure if this is desirable for dask; however, I thought it might be worth pointing out.

Reply:
:+1: thanks, I'll add this to the list.